### PR TITLE
fix libpng wrong function call

### PR DESCRIPTION
**Pull request by bijoshtj**
This change is [Reviewable](https://reviewable.io/reviews/hashcube/native-android/6)

### CODE REVIEW

**Comment by rampr** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/native-android/6#-:-KQyMoUeEnGfbRAgWvVe:bnfp4nl)_

File: TeaLeaf/jni/libpng/pngget.c [hunk 1]
```diff
@@ -862,9 +862,10 @@ png_get_sCAL_fixed(png_structp png_ptr, png_const_infop info_ptr,
    {
       *unit = info_ptr->scal_unit;
       /*TODO: make this work without FP support */
-      *width = png_fixed(png_ptr, atof(info_ptr->scal_s_width), "sCAL width");
-      *height = png_fixed(png_ptr, atof(info_ptr->scal_s_height),
-         "sCAL height");
+      *width = png_fixed(png_ptr, strtod(info_ptr->scal_s_width, NULL),
```
Comment (jishnu7) [inline TeaLeaf/jni/libpng/pngget.c:865]: `atof` is same as `strtod(string, NULL)` right?

**Comment by bijoshtj** [comment]
Some functions in stdlib.h (atof, srand, etc) used to be static inlined functions before android-21 and start to exist in libc.so from android-21.
It looks like your app is compiled with android-21 but run on pre-21 devices.  It may not work because app uses functions don't exist in previous device.

@jishnu7

File: TeaLeaf/jni/libpng/pngget.c (hunk 1 above)
Comment (bijoshtj) [reply TeaLeaf/jni/libpng/pngget.c:865]: yes but atof wont be present in devices with less than android-21 if native target is >=21

**Comment by jishnu7** [comment]
LGTM.

we should sent this to upstream.

**Pull request merged**